src/coreclr/System.Private.CoreLib/src/System/Delegate.CoreCLR.cs

```diff
@@ -435,7 +435,13 @@ private bool BindToMethodInfo(object? target, IRuntimeMethodInfo method, Runtime
         private static MulticastDelegate InternalAlloc(RuntimeType type)
         {
             Debug.Assert(type.IsAssignableTo(typeof(MulticastDelegate)));
-            return Unsafe.As<MulticastDelegate>(RuntimeTypeHandle.InternalAlloc(type));
```

Comment:
Rather than introduce a temp variable, in the case that we are returning the result, just surround the `return` statement directly with an unsafe block.